### ISSUE TITLE
Disable javascript validation

### ISSUE THREAD

**Issue by amiralies** [theia-ide/typescript-language-server#209]
I use this server alongside flow lsp. in vscode it's possible to set `javascript.validate.enable` to `false` so tsserver doesn't report that type annoations are not allowed in js files.
is there anyway to achieve this?
in general can we have tsserver options supported in this server?

**Comment by ybbond** [comment]
👍  I am looking for this feature too

**Comment by rchl** [comment]
Just to understand your use case better... You want to run the Flow server on JS files and simultaneously this server on the TS files?

**Comment by ybbond** [comment]
> Just to understand your use case better... You want to run the Flow server on JS files and simultaneously this server on the TS files?

No, I'd like to run both the flow server and ts server on the JS files. Flow will handle general linting, and ts will handle the 'rest'. For example, a package has no flow-typed definition, but it has *.d.ts definition. So TS will be handy for that particular package.

I am using this server in Sublime's LSP-typescript. In VSCode, setting `javascript.validate.enable` to `false` prevent TS validation error such as `'types' can only be used in a .ts file`. (ref: [VSCode Flow extension page on Ms Store](https://marketplace.visualstudio.com/items?itemName=flowtype.flow-for-vscode#setup))

I don't know amiralies's use case, but I assume same as mine.

**Comment by amiralies** [comment]
Yeah same.

**Comment by lvmtam** [comment]
i am looking for this option. hope it will be available soon

**Comment by predragnikolic** [comment]
Do you really need to disable all diagnostics in js files?

There is a option to disable specific diagnostics with the following workspace setting `diagnostics.ignoredCodes`. See https://github.com/typescript-language-server/typescript-language-server/pull/272#issuecomment-1293828140

**Comment by YangShaoyue** [comment]
Same, I use eslint instead, tsserver is installed in Neovim Mason, javascript validation is enabled default, the hint message showing like this:
![图片](https://user-images.githubusercontent.com/12428824/213338939-e62cb5db-8e5c-424b-9fba-6410b2835490.png)

It's a little annoying and I don't know how to disable it.

**Comment by polsust** [comment]
Hey! Did someone find a solution to this ? 

I don't understand how to implement @predragnikolic solution

**Comment by kocv59** [comment]
While it's possible to disable this in VSCode, it's not an option in Neovim. It would be helpful to allow the setting of `diagnostics.ignoredCodes: true` to ignore all diagnostic codes in Neovim.
